jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt

```diff
@@ -152,6 +171,17 @@ abstract class LambdaBuilder {
             zipLocation.toPath()
         }
 
+    /**
+     * Returns the build directory of the project. Create this if it doesn't exist yet.
+     */
+    private fun getOrCreateBuildDirectory(project: Project): File =
+        WriteAction.computeAndWait<File, Throwable> {
+            val projectBasePath = project.basePath ?: throw IllegalStateException("Invalid project with no base path!")
```

Comment:
I think Project.guessProjectDir() may be better

Reply:
Also use message() for the exception

Reply:
Actually since we know the module, why are we not using the module content root?

Reply:
The implementation of `getBasePath` and `guessProjectDir` are different. `getBasePath` uses `Registry` and looks for `.idea` folder for the project base path. `guessProjectDir` uses modules content roots which as it describes is not always accurate.

I think here we still want to create this folder at the same level to `.idea` instead of in the `HelloWorldFunction` folder which is the model content root.

Reply:
Discussed offline. Decided to use the first content root of the module as the root folder of the build.

Reply:
> Also use message() for the exception

Done.